kratos/utilities/geometry_utilities.h

```diff
@@ -400,6 +401,40 @@ class GeometryUtils
 
     }
 
+    /**
+     * Calculate the exact distances to the plane interface defined by a set 
+     * of initial distances.
+     * @param ThisGeometry Geometry can be either a triangle or a tetrahedra
+     * @param Distances The distances which define the isosurface as input.
+     * Same argument is used to give the calculated exact distances back
+     */
+    template<std::size_t TSize>
+    static void CalculateExactDistancesToPlane(
+        Element::GeometryType& rThisGeometry,
+        array_1d<double, TSize>& rDistances)
+    {
+        array_1d<Point, TSize> intersection_points;
+        int number_of_intersection_points = CalculateTetrahedraIntersectionPoints(rThisGeometry, rDistances, intersection_points);
+
+		if(number_of_intersection_points == 0) {
+			KRATOS_WARNING("GeometryUtilities") << "Warning: The intersection with interface hasn't found! The distances are" << rDistances << std::endl;
+		}
+        else {
+            BoundedMatrix<double,TSize,TSize-1> DN_DX;
+            array_1d<double, TSize> N;
+            double volume;
+            GeometryUtils::CalculateGeometryData(rThisGeometry, DN_DX, N, volume);
+            array_1d<double, TSize-1> distance_gradient = prod(trans(DN_DX), rDistances);
+            distance_gradient /= norm_2(distance_gradient);
```

Comment:
question: what do you do in the case the distance is all 0?

Reply:
hmm in that case there are no intersection points and a warning is thrown...

Reply:
@RiccardoRossi I could add an error if the norm of the distance gradient is lower than a value like 1-e-15.. what do yo think?

Reply:
I don't think we should forbid the distance to be zero. I'd rather set a small tolerance just to avoid division by 0.